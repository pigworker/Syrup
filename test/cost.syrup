not(<Bit>) -> <Bit>
not(X) = nand(X,X)

and(<Bit>, <Bit>) -> <Bit>
and(X,Y) = not(nand(X,Y))

or(<Bit>, <Bit>) -> <Bit>
or(X,Y) = nand(not(X),not(Y))

xor(<Bit>, <Bit>) -> <Bit>
xor(X,Y) = or(and(X,not(Y)),and(not(X),Y))

cost not
cost [] and
cost or
cost [] xor
cost [and,or,not] xor