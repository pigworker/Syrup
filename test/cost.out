not is defined.

and is defined.

or is defined.

xor is defined.

eq is defined.

one is defined.

mux is defined.

xor2 is defined.

not's cost is as follows:
  1 copy of nand

and's cost is as follows:
  2 copies of nand

or's cost is as follows:
  3 copies of nand

or's cost is as follows:
  1 copy of or

or's cost is as follows:
  1 copy of nand
  2 copies of not

xor's cost is as follows:
  9 copies of nand

xor's cost is as follows:
  2 copies of and
  2 copies of not
  1 copy of or

eq's cost is as follows:
  2 copies of and
  3 copies of not
  1 copy of or

zero's cost is as follows:
  1 copy of zero

one's cost is as follows:
  1 copy of nand
  1 copy of zero

one's cost is as follows:
  1 copy of not
  1 copy of zero

mux's cost is as follows:
  2 copies of and
  1 copy of not
  1 copy of or

xor2's cost is as follows:
  1 copy of mux
  1 copy of nand

